src/hyp3_isce2/insar_stripmap.py

```diff
@@ -24,31 +32,84 @@
     os.environ['PATH'] = str(ISCE_APPLICATIONS) + os.pathsep + os.environ['PATH']
 
 
-def insar_stripmap(reference_scene: str, secondary_scene: str) -> Path:
+# TODO update docstring description
+def insar_stripmap(user: str, password: str, reference_scene: str, secondary_scene: str) -> Path:
     """Create an interferogram
 
     This is a placeholder function. It will be replaced with your actual scientific workflow.
 
     Args:
+        user: Earthdata username
+        password: Earthdata password
         reference_scene: Reference scene name
         secondary_scene: Secondary scene name
 
     Returns:
         Path to the output files
     """
+    session = asf_search.ASFSession().auth_with_creds(user, password)
+
+    reference_product, secondary_product = asf_search.search(
+        granule_list=[reference_scene, secondary_scene],
+        processingLevel=asf_search.L1_0,
+    )
+    assert reference_product.properties['sceneName'] == reference_scene
+    assert secondary_product.properties['sceneName'] == secondary_scene
+    products = (reference_product, secondary_product)
 
-    raise NotImplementedError('This is a placeholder function. Replace it with your actual scientific workflow.')
+    polygons = [Polygon(product.geometry['coordinates'][0]) for product in products]
+    insar_roi = polygons[0].intersection(polygons[1]).bounds
 
-    product_file = Path("product_file_name.zip")
-    return product_file
+    dem_path = download_dem_for_isce2(insar_roi, dem_name='glo_30', dem_dir=Path('dem'), buffer=0)
+
+    urls = [product.properties['url'] for product in products]
+    asf_search.download_urls(urls=urls, path=os.getcwd(), session=session, processes=2)
+
+    zip_paths = [product.properties['fileName'] for product in products]
+    for zip_path in zip_paths:
+        with zipfile.ZipFile(zip_path, 'r') as zip_file:
+            zip_file.extractall()
+        os.remove(zip_path)
+
+    reference_image = get_product_file(reference_product, 'IMG-')
+    reference_leader = get_product_file(reference_product, 'LED-')
+
+    secondary_image = get_product_file(secondary_product, 'IMG-')
+    secondary_leader = get_product_file(secondary_product, 'LED-')
+
+    config = stripmapapp.StripmapappConfig(
+        reference_image=reference_image,
+        reference_leader=reference_leader,
+        secondary_image=secondary_image,
+        secondary_leader=secondary_leader,
+        roi=insar_roi,
+        dem_filename=str(dem_path),
+    )
+    config_path = config.write_template('stripmapApp.xml')
+
+    stripmapapp.run_stripmapapp(start='startup', end='geocode', config_xml=config_path)
+
+    # TODO is this still needed?
+    #stripmapapp.run_stripmapapp(start='rubber_sheet_range', end='geocode', config_xml=config_path)
```

Comment:
Is this commented-out `run_stripmapapp` function call still potentially useful for any reason? Otherwise, we should delete it.
```suggestion
```

Reply:
Yes, it can be removed. I changed the start and end to fix some errors but now it's not necessary.